app/javascript/packs/unpublishPostModal.jsx

```diff
@@ -0,0 +1,149 @@
+import { h, render } from 'preact';
+import PropTypes from 'prop-types';
+import { request } from '../utilities/http';
+import { ButtonNew as Button } from '@crayons';
+import RemoveIcon from '@images/x.svg';
+
+async function confirmAdminUnpublishPost(id, username, slug) {
+  try {
+    const response = await request(`/articles/${id}/admin_unpublish`, {
+      method: 'PATCH',
+      body: JSON.stringify({ id, username, slug }),
+      credentials: 'same-origin',
+    });
+
+    const outcome = await response.json();
+
+    /* eslint-disable no-restricted-globals */
+    if (outcome.message == 'success') {
+      window.top.location.assign(`${window.location.origin}${outcome.path}`);
+    } else {
+      top.addSnackbarItem({
+        message: `Error: ${outcome.message}`,
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+
+  toggleUnpublishPostModal();
+}
+
+/**
+ * Shows or hides the flag user modal.
+ */
+export function toggleUnpublishPostModal() {
+  const modalContainer = top.document.getElementsByClassName(
+    'unpublish-post-modal-container',
+  )[0];
+  modalContainer.classList.toggle('hidden');
+
+  if (!modalContainer.classList.contains('hidden')) {
+    top.window.scrollTo(0, 0);
+    top.document.body.style.height = '100vh';
+    top.document.body.style.overflowY = 'hidden';
+  } else {
+    top.document.body.style.height = 'inherit';
+    top.document.body.style.overflowY = 'inherit';
+  }
+}
+
+/**
+ * Initializes the Unpublish Post modal for the given article ID, author username and article slug.
+ *
+ * @param {number} articleId
+ * @param {string} authorUsername
+ * @param {string} articleSlug
+ */
+export function initializeUnpublishPostModal(
+  articleId,
+  authorUsername,
+  articleSlug,
+) {
+  // Check whether context is ModCenter or Friday-Night-Mode
+  const modContainer = document.getElementById('mod-container');
+
+  if (!modContainer) {
+    return;
+  }
+
+  render(
+    <UnpublishPostModal
+      articleId={articleId}
+      authorUsername={authorUsername}
+      articleSlug={articleSlug}
+    />,
+    document.getElementsByClassName('unpublish-post-modal-container')[0],
+  );
+}
+
+/**
+ * A modal for unpublishing a post. This can be used in the moderation center
+ * or on an article page.
+ *
+ * @param {number} props.articleId ID of the article to be unpublished.
+ * @param {string} props.authorUsername Username of the article's author.
+ * @param {string} props.articleSlug Slug of the article to be unpublished.
+ */
+export function UnpublishPostModal({ articleId, authorUsername, articleSlug }) {
```

Comment:
@aitchiss Hey friend! 👋🏾 

I think I need your help greatly on this PR. So when I started building out the modal for unpublishing a post, I followed the existing "Flag User" implementation. Later on, I noticed that the Flag User modal does NOT use our `<Modal />` component. 🤦🏾‍♀️ 

 So I attempted to use the `Modal` component in my current PR. I noticed a few issues:
- In Storybook, the button that triggers the Modal and the Modal itself are in the same page, and they are both imported components. In my use case, the trigger Button and Modal are on the same page, BUT the Button is in Rails (erb file). I was unclear how to send the `setIsModalOpen(true)` useState onClick from this Rails button. 😓 
- The modals in Storybook do not have buttons inside them, to take action from within the Modal. In my use case, I need a destructive button within the modal.

What are your thoughts about using the `Modal` component in this PR? If you think we can and should, I would love to pair with you and learn how you connect the Rails and React parts. 😅 If you think my current implementation is alright, then good! However, with my current code, the darkened modal backdrop does not show up. I noticed that the backdrop itself is now a component lol.

Thank you for reading this long comment! ❤️